File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> m;

        for(int i = 0; i < nums.size(); i++){
            int comp = target - nums[i];
            if(m.contains(comp)){
                return {m[comp], i};
            }
            m[nums[i]] = i;
        }

        return {};
    }
};

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)

    // for loop range gives access to both index and element
    for i, n := range nums{
        // returns value if ok
        if v, ok := m[target - n]; ok{
            return []int{i, v}
        }
        m[n] = i
    }
    return nil
}